src/python/pants/pantsd/process_manager.py

```diff
@@ -448,32 +456,43 @@ def daemonize(self, pre_fork_opts=None, post_fork_parent_opts=None, post_fork_ch
     daemons. Having a disparate umask from pre-vs-post fork causes files written in each phase to
     differ in their permissions without good reason - in this case, we want to inherit the umask.
     """
+    fork_context = fork_context or self._noop_fork_context
+
+    def double_fork():
```

Comment:
Is it work explaining in this context why double forking is necessary?

Reply:
It's explained in the comment above.

Reply:
I missed that! Sorry!